app/src/main/java/com/example/grutor/Fragments/LessonsFragment.java

```diff
@@ -40,9 +44,11 @@ public class LessonsFragment extends Fragment {
     private static final String KEY_CREATED_AT_QUERY = "createdAt";
     protected StudentMatcher matching;
     RecyclerView rvLessons, rvMatches;
-    LessonAdapter lessonsAdapter;
+    LessonAdapter lessonsAdapter, studentAdapter, tutorAdapter;
```

Comment:
Looks like you create these adapter and call setAdapter() then. you really need them to be class level variables?